llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -500,6 +500,44 @@ static bool isSelect01(const APInt &C1I, const APInt &C2I) {
   return C1I.isOne() || C1I.isAllOnes() || C2I.isOne() || C2I.isAllOnes();
 }
 
+/// Try to simplify a select instruction when the user of its select user
+/// indicates the condition.
+static bool simplifySeqSelectWithSameCond(SelectInst &SI,
+                                          const SimplifyQuery &SQ,
+                                          InstCombinerImpl &IC) {
+  Value *CondVal = SI.getCondition();
+  if (match(CondVal, m_ImmConstant()))
+    return false;
```

Comment:
Isn't it handled by `simplifySelectInst`?

Reply:
Yes, it isn't.
[#85663](https://github.com/llvm/llvm-project/pull/85663) is a more general improvement by nikic, and it has some compile time regression, so I continue on this solution.

Reply:
https://github.com/llvm/llvm-project/blob/a61252419779a6d4a5ebf71e7e2fc4adc75cfddd/llvm/lib/Analysis/InstructionSimplify.cpp#L4767-L4771

Can you provide a test to show that the early exit here is necessary?

Reply:
Thanks, I delete the early exit checking.
I added it just because I thought it was an optimized expression and didn't need to be optimized.